action.yml

```diff
@@ -68,13 +67,30 @@ inputs:
     description: The setting for `RUST_BACKTRACE` (see https://doc.rust-lang.org/std/backtrace/index.html#environment-variables)
     required: false
 
-
 runs:
   using: composite
   steps:
     - name: Install Nix
       shell: bash
       run: |
+        IDENTIFIER=$(basename $GITHUB_ACTION_PATH)
+        TAG=$(echo "$IDENTIFIER" | { perl -nle'print if m{^v(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$}' || true; })
```

Comment:
Yeah doing cross platform regex can get a bit ugly because Mac doesn't package a nice grep.

In most cases, yes, splitting the action out is a good idea.

Here are some considerations the other way:

* If we split the action out, every update to the planner settings will necessitate a similar update to the action
* Any changes to how the binary is invoked will require an action update as well
* It would be quite confusing to have `v1.0.0` of the action necessary for say, versions `v1.0.0` through `v1.5.0` of the installer, then have some action `v1.0.1` match versions `v1.5.1` through `v2.1.0` of the installer, so we'd need to figure out some way to handle that (which would likely be having an action version for each nix-installer version?)
* Adding things like `nix-installer-tag` or `nix-installer-pr` options to the action would even further encourage situations where users mismatch their action version to the installer version.
* Any work involving the action and binary would require coordinating over two repos. While not so common anymore, this was frequently happening earlier in development.

In this case, it's the action's job is to invoke the binary as directly as possible. The binary is essentially supposed to disappear and the user shouldn't think about it at all after. They're not using the action to install the binary, they're using the action to run the binary to install Nix.

I am not sure the decision is so clear.

Further, here are some benefits from keeping them together:

* Users get a slightly more canonical `determinatesystems/nix-installer` instead of some `determinatesystems/nix-installer-action`
* Users can pretty directly pick a nix-installer version with `determinatesystems/nix-installer@v1.0.0`
* Users are always guaranteed to get an action that works with their nix-installer version
* We never need to coordinate across repos for action/binary work
* Our CI can keep directly using an in-repo action for self testing

Reply:
Yeah, I think it would be reasonable keep the action in a different repo.

I can imagine other circumstances / integrations where we may want to publish and revise them separately, too, like buildkite and other CI services. Or Mosyle integrations. These will all have similar API concerns as the action <-> installer options and we'll just need to figure it out, and be aware of what we're doing when we break the API. I think trying to keep them in the same repo is going to cause (and is already causing) a lot of headache that we don't actually need to sign up for.

I also think we probably won't be breaking the interface very much, since we are expecting people to be integrating this into their automation on a regular basis. I think the risk of needing highly coordinated changes is relatively low.

Reply:
Cool let's do that then. :)